logback-classic/src/main/java/ch/qos/logback/classic/encoder/JsonEncoder.java

```diff
@@ -264,40 +268,66 @@ private void appendThrowableProxy(StringBuilder sb, String attributeName, IThrow
+        } else {
+            appendSTEPArray(sb, itp.getStackTraceElementProxyArray(), itp.getCommonFrames());
+        }
 
         if (itp.getCommonFrames() != 0) {
             sb.append(VALUE_SEPARATOR);
             appenderMemberWithIntValue(sb, COMMON_FRAMES_COUNT_ATTR_NAME, itp.getCommonFrames());
         }
 
-        IThrowableProxy cause = itp.getCause();
-        if (cause != null) {
-            sb.append(VALUE_SEPARATOR);
-            appendThrowableProxy(sb, CAUSE_ATTR_NAME, cause);
-        }
+        if (!withOriginalStackTrace) {
+            IThrowableProxy cause = itp.getCause();
+            if (cause != null) {
+                sb.append(VALUE_SEPARATOR);
+                appendThrowableProxy(sb, CAUSE_ATTR_NAME, cause);
+            }
 
-        IThrowableProxy[] suppressedArray = itp.getSuppressed();
-        if (suppressedArray != null && suppressedArray.length != 0) {
-            sb.append(VALUE_SEPARATOR);
-            sb.append(QUOTE).append(SUPPRESSED_ATTR_NAME).append(QUOTE_COL);
-            sb.append(OPEN_ARRAY);
-            boolean first = true;
-            for (IThrowableProxy suppressedITP : suppressedArray) {
-                if (first) {
-                    first = false;
-                } else {
-                    sb.append(VALUE_SEPARATOR);
+            IThrowableProxy[] suppressedArray = itp.getSuppressed();
+            if (suppressedArray != null && suppressedArray.length != 0) {
+                sb.append(VALUE_SEPARATOR);
+                sb.append(QUOTE).append(SUPPRESSED_ATTR_NAME).append(QUOTE_COL);
+                sb.append(OPEN_ARRAY);
+                boolean first = true;
+                for (IThrowableProxy suppressedITP : suppressedArray) {
+                    if (first) {
+                        first = false;
+                    } else {
+                        sb.append(VALUE_SEPARATOR);
+                    }
+                    appendThrowableProxy(sb, null, suppressedITP);
                 }
-                appendThrowableProxy(sb, null, suppressedITP);
+                sb.append(CLOSE_ARRAY);
             }
-            sb.append(CLOSE_ARRAY);
         }
 
         sb.append(CLOSE_OBJ);
 
     }
 
+    private static String getOriginalStackTrace(IThrowableProxy throwableProxy) {
+        StringBuilder sb = new StringBuilder();
+        getOriginalStackTrace(throwableProxy, sb, 0);
+        return sb.toString();
+    }
+
+    private static void getOriginalStackTrace(IThrowableProxy throwable, StringBuilder sb, int depth) {
+        if (throwable == null) {
+            return;
+        }
+        if (depth > 0) {
+            sb.append("Caused by: ");
+        }
+        sb.append(throwable.getClassName()).append(": ").append(throwable.getMessage()).append(NEWLINE);
+        for (StackTraceElementProxy step : throwable.getStackTraceElementProxyArray()) {
+            sb.append(TAB).append(step).append(NEWLINE);
```

Comment:
```suggestion
            sb.append(TAB).append(step.getSTEAsString()).append(NEWLINE);
```

Ensure that we use the Stack Trace Element toString even if `StackTraceElementProxy.toString()` gets modified for debugging purpose